File: Microsoft Office Developer Documentation Team/Outlook-Social-Connector-2013-Provider-templates-17387/[C++]-Outlook Social Connector 2013 Provider templates/C++/OSCProfile.cpp
// OSCProfile.cpp : Implementation of COSCProfile

#include "stdafx.h"
#include "OSCProfile.h"

// Not supported since OSC 2013
STDMETHODIMP COSCProfile::GetActivities(DATE startTime, BSTR * activities)
{
	return OSC_E_NOT_IMPLEMENTED;
}

// Return an XML stream of the person's friends/colleagues and their details
STDMETHODIMP COSCProfile::GetFriendsAndColleagues(BSTR * personsCollection)
{
	return OSC_E_NOT_IMPLEMENTED;
}

// Return an array of IDs of the person's friends/colleagues
STDMETHODIMP COSCProfile::GetFriendsAndColleaguesIDs(SAFEARRAY * * friendsIDs)
{
	return OSC_E_NOT_IMPLEMENTED;
}

// Get the person's picture in a byte array
STDMETHODIMP COSCProfile::GetPicture(SAFEARRAY * * picture)
{
	return OSC_E_NOT_IMPLEMENTED;
}

// Not supported
STDMETHODIMP COSCProfile::GetStatus(BSTR * status)
{
	return OSC_E_NOT_IMPLEMENTED;
}

// Get an XML stream of the person's details
STDMETHODIMP COSCProfile::GetDetails(BSTR * details)
{
	return OSC_E_NOT_IMPLEMENTED;
}

// Determines whether the specified users are friends
STDMETHODIMP COSCProfile::AreFriendsOrColleagues(SAFEARRAY * userIDs, SAFEARRAY * * results)
{
	return OSC_E_NOT_IMPLEMENTED;
}


// Not supported since OSC 2013
STDMETHODIMP COSCProfile::GetActivitiesOfFriendsAndColleagues(DATE startTime, BSTR * activitiesCollection)
{
	return OSC_E_NOT_IMPLEMENTED;
}

// Not supported
STDMETHODIMP COSCProfile::SetStatus(BSTR status)
{
	return OSC_E_NOT_IMPLEMENTED;
}

File: Microsoft Office Developer Documentation Team/Outlook-Social-Connector-2013-Provider-templates-17387/[C#]-Outlook Social Connector 2013 Provider templates/C#/OSCProvider_CS/OSCProfile.cs
using System;
using System.Diagnostics;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OutlookSocialProvider;

namespace OSCProvider_CS
{
    class OSCProfile : OSCPerson, OutlookSocialProvider.ISocialProfile 
    {
        #region ISocialProfile Members

        public bool[] AreFriendsOrColleagues(string[] userIDs)
        {
            //To-do: return an array indicating persons in userIDs are friends
            return null;
        }

        public string GetActivitiesOfFriendsAndColleagues(DateTime startTime)
        {
            //Not supported since OSC 2013
            return string.Empty;
        }

        public void SetStatus(string Status)
        {
            //Not supported
        }

        #endregion

        #region ISocialPerson members
        public new string GetActivities(DateTime startTime)
        {
            //Not supported since OSC 2013
            return string.Empty;
        }


        public new string GetDetails()
        {
            //To-do: Implement GetDetails
            return string.Empty;
        }

        public new string GetFriendsAndColleagues()
        {
            //To-do: Implement GetFriendsAndColleagues
            return string.Empty;
        }

        public new string[] GetFriendsAndColleaguesIDs()
        {
            //To-do: Implement GetFriendsAndColleaguesIDs
            return null;
        }

        public new byte[] GetPicture()
        {
            //To-do: Implement GetPicture
            return null;
        }

        public new string GetProfileUrl()
        {
            //To-do: Implement GetDetails
            return string.Empty;
        }

        public new string GetStatus()
        {
            //Not supported 
            throw new NotImplementedException();
        }


        #endregion
    }
}
